src/components/FilterCard.vue

```diff
@@ -83,42 +79,48 @@ export default {
       default: true,
     },
   },
-  data() {
-    return {
-      filter: {
-        value: '',
-        input: '',
-        wildcard: true,
-      },
-    }
+  setup() {
+    const { filter_input, filter_value, filter_wildcard, page_size, page_number, clearFilter } =
+      useQueryParams()
+    return { filter_input, filter_value, filter_wildcard, page_size, page_number, clearFilter }
   },
   computed: {
     isWildcardOption() {
-      return this.filterOptions.filter(({ value }) => value == this.filter.value)[0]?.wildcard
+      return this.filterOptions.filter(({ value }) => value == this.filter_value)[0]?.wildcard
+    },
+  },
+  watch: {
+    /*
+      page_number and page_size:
+      We ensure the page is paginated before fetching data.
+      This catches an edge case where the watchers are still active when navigating
+      between paginated and unpaginated routes
+    */
+    page_size() {
+      this.$route.meta?.paginated ? this.getData() : ''
+    },
+    page_number() {
+      this.$route.meta?.paginated ? this.getData() : ''
```

Comment:
Ok, I got it. I think, in that way it make more sense for the watchers to be here rather than in pagination. Again, as you mentioned it doesn't bring any specific advantage by moving this fetching to one generic component to another.